ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -854,84 +949,73 @@ namespace OrdinalAxis {
          * @param {number} val
          *        The linear abstracted value.
          *
-         * @param {boolean} [fromIndex]
+         * @param {boolean} [isInside]
          *        Translate from an index in the ordinal positions rather than a
          *        value.
          *
          * @return {number}
          */
-        axisProto.lin2val = function (
-            val: number,
-            fromIndex?: boolean
-        ): number {
-            let axis = this,
+        axisProto.lin2val = function (val: number, isInside: boolean|undefined): number {
+            const axis = this,
                 ordinal = axis.ordinal,
-                ordinalPositions = ordinal.positions,
-                ret;
+                ordinalPositions = ordinal.positions; // for the current visible range
+            let extendedOrdinalPositions = this.ordinal.extendedOrdinalPositions;
 
-            // the visible range contains only equally spaced values
+            // The visible range contains only equally spaced values.
             if (!ordinalPositions) {
-                ret = val;
+                return val;
+            }
 
-            } else {
+            // If the value is not inside the plot area,
+            // use the extended positions.
+            // (array contains also points that are currently out of range).
+            if (!isInside) {
+                // When iterating for the first time,
+                // get the extended ordinal positional and assign them.
+                if (axis.ordinal && !extendedOrdinalPositions) {
+                    extendedOrdinalPositions = axis.ordinal.getExtendedPositions();
+                    this.ordinal.extendedOrdinalPositions = extendedOrdinalPositions;
+                }
+
+                if (ordinalPositions && extendedOrdinalPositions && extendedOrdinalPositions.length) {
```

Comment:
Do we need a check for `ordinalPositions`? In this case, we use only `extendedOrdinalPositions`